Update to emoji-data 4.0.0

liliakai opened the pull request:
Add support for :clown_face: :cowboy_hat_face: :nauseated_face: :sneezing_face:  and more. Because everybody wants more emoji!

Before/After
<img width="1434" alt="schermafbeelding 2017-11-19 om 3 03 43 pm" src="https://user-images.githubusercontent.com/1039940/32996712-0a9d4706-cd3b-11e7-8182-b5aa53efebb2.png">

rainerzufall commented:
@liliakai thanks so much! As a Signal (Desktop beta) user I wanted to ask for an update in your own repository, but since you are already working on updating the emojis here - i will ask here - have you by any chance missed the newest release of the `iamcal/emoji-data`? `emoji-datasource v4.0.0` has been released only two days ago:  https://github.com/iamcal/emoji-data/releases/tag/v4.0.0 & https://www.npmjs.com/package/emoji-datasource Maybe it is possible to not "only" upgrade to 3.0.0 but to directly go to 4.0.0? Thanks in any case for your work on Signal! :)

liliakai committed:
Update deps to emoji-datasource 4.0.x

liliakai committed:
Update category name map

Category names changed in emoji-data 4.0.x

diff --git a/src/map.js b/src/map.js
index c2f4dc8..45369f9 100644
--- a/src/map.js
+++ b/src/map.js
@@ -4,14 +4,14 @@ const SIZE = Constant.SIZE;
 const CATEGORY = Constant.CATEGORY;

 const categoryDataMap = {
-  Activity: CATEGORY.ACTIVITY,
-  Flags: CATEGORY.FLAGS,
-  Foods: CATEGORY.FOODS,
-  Nature: CATEGORY.NATURE,
-  Objects: CATEGORY.OBJECTS,
-  People: CATEGORY.PEOPLE,
-  Places: CATEGORY.PLACES,
-  Symbols: CATEGORY.SYMBOLS
+  "Activities": CATEGORY.ACTIVITY,
+  "Flags": CATEGORY.FLAGS,
+  "Food & Drink": CATEGORY.FOODS,
+  "Animals & Nature": CATEGORY.NATURE,
+  "Objects": CATEGORY.OBJECTS,
+  "Smileys & People": CATEGORY.PEOPLE,
+  "Travel & Places": CATEGORY.PLACES,
+  "Symbols": CATEGORY.SYMBOLS
 };

 const categoryNameMap = {

liliakai committed:
Fix background position calculation

Sprite format changed slightly in emoji-datasource 4.0.x

> Added gaps between images in spritesheets to avoid bleed when displaying them
> zoomed - a 1px border around each image at all sizes

https://github.com/iamcal/emoji-data/blob/v4.0.0/CHANGES.md

liliakai commented:
Thanks, @rainerzufall. Updated to emoji-datasource v4.0.0.

The emoji datasource v3.0.0 work is still available [here](https://github.com/liliakai/emoji-panel/tree/emoji3.0), but the changes needed were made obsolete by v4.0.0, so I thought it would be cleaner to drop those commits entirely from this PR.
